static/app/views/alerts/rules/metric/ruleConditionsForm.tsx

```diff
@@ -583,7 +583,7 @@ class RuleConditionsForm extends PureComponent<Props, State> {
                   organization={organization}
                   placeholder={this.searchPlaceholder}
                   onChange={onChange}
-                  query={model.getValue('query')}
+                  query={initialData.query}
```

Comment:
@Neo-Zhixing so it's intentional that we don't want this to change when the for mchagnes?

Reply:
Yeah, turns out this is the easiest way to fix this. Kind of an anti-pattern here, generally for controlled components you want to update the props on change.